### ISSUE TITLE
[4.0][3.10][PHP 8] Joomla\CMS\Filter\InputFilter::isSafeFile() tries to check non-existent uploads

### ISSUE THREAD

**Issue by OctavianC** [joomla/joomla-cms#32277]
### Steps to reproduce the issue
When file uploads are optional, isSafeFile() still tries to check files that haven't been uploaded.
This is more obvious when using PHP 8 since fopen($path) will throw an error if $path is empty.
The only reason the below steps are needed is to easily illustrate the issue since I couldn't find a single place in Joomla! that has optional uploads.

- Edit `administrator/components/com_media/views/tmpl/default.php`
- Search for `<input required type="file" id="upload-file" name="Filedata[]" multiple />`
- Remove the required attribute so that it reads `<input type="file" id="upload-file" name="Filedata[]" multiple />`
- Additionally you can simply change the required attribute by using the Inspector in your browser
- Go to `Content > Media`
- Click on `Upload` on the toolbar
- Don't select a file
- Click on `Start Upload`

### Expected result
Since no upload takes place, the page should refresh with an enqueued message.

### Actual result

Error
An error has occurred.

    0 Path cannot be empty
    JPATH_ROOT\libraries\src\Filter\InputFilter.php:630 

### System information (as much as possible)
PHP 8.0.1

### Additional comments
Error comes from this line in `libraries\src\Filter\InputFilter.php`:
```php
$fp = @fopen($tempName, 'r');`
```

The `$_FILES` array is populated, but `tmp_name` will be an empty string and `error` will be set to 4 which is `UPLOAD_ERR_NO_FILE`. The logic in `isSafeFile()` will still attempt to `fopen()` this empty string which leads to an error in PHP 8. Easiest fix would be:

```php
$fp = strlen($tempName) ? @fopen($tempName, 'r') : false;
```

A more elaborate fix would be to also check the `error` segment of the `$_FILES` array.

**Comment by richard67** [comment]
Closing as having a pull request. Please test #32372 . Thanks in advance.

**Comment by richard67** [comment]
Re-opening since this issue is for both 3.x and 4.0, but PR #32372 fixes it only for 4.0.

**Comment by richard67** [comment]
Pull request for staging is #32374 , for 4.0-dev it's #32372 . Please test. Thanks in advance,
